lib/web.ts

```diff
@@ -166,10 +167,22 @@ export default class AutoConsent {
                 this.addDeclarativeCMP(ruleset);
             });
         }
+
+        if (declarativeRules.compact) {
+            try {
+                const rules = decodeRules(declarativeRules.compact);
+                rules.forEach(this.addDeclarativeCMP.bind(this));
+            } catch (e) {
+                this.config.logs.errors && console.error(e);
+            }
```

Comment:
I think we can move the try..catch block inside the loop here, so that errors in individual rules don't prevent others from being added.

Reply:
The try catch here is for the explicit exception emitted by `decodeRules` if the rules are a newer encoding version. I don't think `addDeclarativeCMP` can throw because it's just calling `new AutoConsentCMP` and appending to a list.